.github/workflows/ci.yml

```diff
@@ -182,6 +184,18 @@ jobs:
           command: clippy
           args: --all-targets --all-features
 
+  typos:
+    name: Check for typos
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+        # Typos check should be pinned to the very specific version
+        # to prevent sudden dictionary updates from making our CI fail
+      - name: Check for typos in the repository
+        uses: crate-ci/typos@v1.12.14
```

Comment:
Are you sure that's a good idea to use a dictionary in CI (esp. require it)?

For example I have a vscode plugin that checks for typos, but when working on, for example, `rustc` it gives a whole lot of false positives (?) wrt abbreviation and local terms.

Reply:
The dictionary is statically embedded into `typos` binary, so there is no network access happening just to make it clear.

`typos` specifically designed to avoid false positives. If there are false positives, they can be added to the dictionary via the config. For example, in our private repo we have just this
```
[default.extend-identifiers]
# Crate name
mke2fs = "mke2fs"
```
If needed some pervasive abbreviations can be added here.

Typos CLI is better than VSCode plugin, because it is integrated into CI, doesn't depend on any particular editor, is extensible, and could even be run on pre-commit hook.

The downside of course is that it is not exhaustive. It does miss typos that are not in its static dictionary, but it is good enough.